platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -1198,3 +1198,155 @@ def get_sfp(self, index):
             return None
 
         return module.get_sfp(sfp_index - 1)
+
+class SmartSwitchChassis(Chassis):
+    def __init__(self):
+        super(SmartSwitchChassis, self).__init__()
+        self.module_initialized_count = 0
+        self.module_name_index_map = {}
+        self.initialize_modules()
+
+    def is_modular_chassis(self):
+        """
+        Retrieves whether the sonic instance is part of modular chassis
+
+        Returns:
+            A bool value, should return False by default or for fixed-platforms.
+            Should return True for supervisor-cards, line-cards etc running as part
+            of modular-chassis.
+            For SmartSwitch platforms this should return True even if they are
+            fixed-platforms, as they are treated like a modular chassis as the
+            DPU cards are treated like line-cards of a modular-chassis.
+        """
+        return False
+
+    ##############################################
+    # Module methods
+    ##############################################
+    def initialize_single_module(self, index):
+        count = self.get_num_modules()
+        if index < 0:
+            raise RuntimeError(f"Invalid index = {index} for module initialization with total module count = {count}")
+        if index >= count:
+            return
+        if not self._module_list:
+            self._module_list = [None] * count
+        if not self._module_list[index]:
+            from .module import DpuModule
+            module = DpuModule(index)
+            self._module_list[index] = module
+            self.module_name_index_map[module.get_name()] = index
+            self.module_initialized_count += 1
+
+    def initialize_modules(self):
+        count = self.get_num_modules()
+        for index in range(count):
+            self.initialize_single_module(index=index)
+
+    def get_num_modules(self):
+        """
+        Retrieves the number of modules available on this chassis
+        On a SmarSwitch chassis this includes the number of DPUs.
+
+        Returns:
+            An integer, the number of modules available on this chassis
+        """
+        return DeviceDataManager.get_dpu_count()
```

Comment:
This seems to be returning only DPUs?

Reply:
Yes, only the DPUs are represented as modules, the switch itself is not a module